### PR TITLE
CSI - bump the current CSI version from v1.16 to v1.17 and update release documents

### PR DESCRIPTION
**Pull request by rahman-lb**
Adding documentation for new CSI release v1.17.0 with Lightbits core v3.12.1

Testing Performed.
- Test from 405 , 622, 626 Passed.
-  Manual testing (WIP)

Issue :  LBM1-34848 

## Release Notes for Version 1.17.0

- **New Features**
  - Updated compatibility with Kubernetes versions, now supporting up to v1.30.
  - Enhanced support for pod tolerations in the CSI Plugin.

- **Improvements**
  - Version updates for the Lightbits CSI Plugin and associated Helm charts.
  - Updated documentation for the Lightbits CSI Plugin deployment and installation processes.
  - Updated compatibility with Lightbits™ software, now supporting version 3.12.1.

- **Changelog**
  - Added detailed CHANGELOG for version 1.17.0, including links to source code and upgrade instructions.

### CODE REVIEW

**Comment by muliby-lb** [comment]
@elada-lb is the PR check ready and we can run it on it?
@rahman-lb since the PR check may not be ready in time, pls check it manually, I'm sure @ronen-lb knows how to do it

**Review by ronen-lb — COMMENTED** [review]
@rahman-lb looks legit, please seem comment from codeRabbit/Myself.
please also mention in PR how was this tested

File: docs/metadata.md [hunk 1]
```diff
@@ -3,12 +3,12 @@ title:  |
...
 
 
-  Lightbits CSI Plugin v1.16.0 Deployment Guide
+  Lightbits CSI Plugin v1.17.0 Deployment Guide
 subtitle: |
 
   LightOS Version: v3.10.1
 
-  Kubernetes Versions: v1.17 - v1.24
+  Kubernetes Versions: v1.17 - v1.30
```
[the remark this answers is not in the document]
Comment (ronen-lb) [reply docs/metadata.md:11]: @rahman-lb seems like support for version 30 was not merged to our master?

File: deploy/helm/lb-csi/values.yaml [hunk 2]
```diff
@@ -1,14 +1,14 @@
 ---
 enableExpandVolume: true
 enableSnapshot: true
-image: "lb-csi-plugin:1.16.0"
+image: "lb-csi-plugin:1.17.0"
```
[the remark this answers is not in the document]
Comment (ronen-lb) [reply deploy/helm/lb-csi/values.yaml:4]: @ron-lb real cool catch by codeRabbit, exactly type of issue easy to miss in reviews (we tend to review only what was changed and not what may also have needed to change).
@rahman-lb please update the above (i pushed a commit for missing changes, ideally should squash but not critical)

[the remark this answers is not in the document]
Comment (ronen-lb) [reply deploy/helm/lb-csi/values.yaml:4]: @coderabbitai thanks for correct feedback. in same file seems like the helm chart version that was also updated from 1.14.0 to 1.15.0 was missed by code changes and this review

**Comment by ronen-lb** [comment]
@rahman-lb please add to ticket how this was verified, we don't currtnly have PR for this, so as muli mentioned, need manual sanity test for this

**Review by ronen-lb — COMMENTED** [review]
Majority of changes seem cool, however need to still:
1. Run manually and verify/update here that was already done.
2. reply regarding support for csi version 30, any changes needed elsewhere here for this?

**Comment by elada-lb** [comment]
> @elada-lb is the PR check ready and we can run it on it? @rahman-lb since the PR check may not be ready in time, pls check it manually, I'm sure @ronen-lb knows how to do it

Not ready yet, I'll need to know which system tests are required to run in the PR check and can prepare it ASAP

**Comment by rahman-lb** [comment]
@elada-lb  you can start with test # 405_lb_csi_k8s_1_30_smoke

**Comment by rahman-lb** [comment]
FYI, I'll be testing the change today, via automated test and some edge cases manually,

File: docs/metadata.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rahman-lb) [reply docs/metadata.md:11]: @ronen-lb  deployment manifest for v25-v30,   https://github.com/LightBitsLabs/los-csi/pull/60

File: docs/src/introduction.md [hunk 3]
```diff
@@ -13,11 +13,11 @@
 
 The Lightbits™ CSI plugin is a software module that implements the management of persistent storage volumes exported by Lightbits™ software, for Container Orchestrator (CO) systems such as Kubernetes and Mesos. In conjunction with the Lightbits™ disaggregated storage solution, the CSI plugin provides a building block for the easy deployment of stateful containerized applications on CO clusters.
 
-The version of the plugin covered by this document implements version 1.2 of the [Container Storage Interface (CSI) Specification](https://github.com/container-storage-interface/spec/blob/v1.2.0/spec.md), and is compatible with Lightbits™ version 3.10.1.
+The version of the plugin covered by this document implements version 1.2 of the [Container Storage Interface (CSI) Specification](https://github.com/container-storage-interface/spec/blob/v1.2.0/spec.md), and is compatible with Lightbits™ version 3.12.1.
```
[the remark this answers is not in the document]
Comment (ronen-lb) [reply docs/src/introduction.md:16]: @rahman-lb ?

File: docs/metadata.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ronen-lb) [reply docs/metadata.md:11]: @rahman-lb why not add as commits to this PR?

[the remark this answers is not in the document]
Comment (rahman-lb) [reply docs/metadata.md:11]: @ronen-lb I wanted to separate the PR because 
   - If something goes wrong in support version from v1.25 -v1.30 , we don't need to change anything on PR  bumping version (except version v.130 mentioned) 
   - Any new CSI release uses previous version's bumping PR as references, and i wanted to keep the reference PR as much clean as possible.  
   -

File: docs/src/introduction.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (rahman-lb) [reply docs/src/introduction.md:16]: fixed.
